src/Eloquent/HasDocumentVersion.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MongoDB\Laravel\Eloquent;
+
+trait HasDocumentVersion
+{
+    /**
+     * Auto call on model instance as booting
+     *
+     * @return void
+     */
+    public static function bootHasDocumentVersion(): void
+    {
+        static::saving(function ($model) {
+          $version = $model->getDocumentVersion();
+
+          if (!$version) {
+              $model->setDocumentVersion(static::getCurrentVersion());
+          }
```

Comment:
`$version` is used only once, not necessary.
```suggestion
          if (! $model->getDocumentVersion()) {
              $model->setDocumentVersion(static::getCurrentVersion());
          }
```

What's the behavior when the stored document doesn't contain the `schema_version` field.  It should be considered `0` and `$model->migrateDocumentVersion` should be called.